tock/hours/models.py

```diff
@@ -138,15 +138,95 @@ def get_message_enabled_display(self):
         return 'Message Enabled'
 
     def get_fiscal_year(self):
-        """Determines the Fiscal Year (Oct 1 - Sept 31) of a given
+        """Determines the Fiscal Year (Oct 1 - Sept 30) of a given
             ReportingPeriod. Oct, Nov, Dec are part of the *next* FY """
         next_calendar_year_months = [10, 11, 12]
-        if self.start_date.month in next_calendar_year_months:
+        # When this reporting period spans two fiscal years
+        if self.start_date.month == 9 and self.end_date.month == 10:
+            # if start date is 9/27 or earlier, it means there's more September
+            # than October in this period, so it belongs to FY of September
+            if self.start_date.day <= 27:
+                return self.start_date.year
+            # Else there's more October days, it belongs to the FY of October
+            else:
+                return self.start_date.year + 1
+
+        elif self.start_date.month in next_calendar_year_months:
             fiscal_year = self.start_date.year + 1
             return fiscal_year
         else:
             return self.start_date.year
 
+
+    @staticmethod
+    def get_fiscal_year_start_date(fiscal_year):
+        """
+        Normally the FY start date is 10/1 of the previous year, but since our
+        system makes use of reporting period, we cannot just easily use this
+        date.  So in this system, during the week that spans two fiscal years,
+        that week will belong to the previous year if there's more September
+        days than October. And more October days means it belongs to this year.
+        i.e. 10/1/2015 is on a Thursday, this means it belongs to FY 2015 since
+        there is more September days that week (Sun to Wed), so the start date
+        of FY 2016 is 10/04/2015 (that coming Sunday).
+        """
+        first_date = datetime.date(fiscal_year - 1, 10, 1)
+        # converting definition of weekday for easier calculation,
+        # so Sun = 0, Mon = 1, ... Sat = 6
+        normalized_weekday = (first_date.weekday() + 1) % 7
```

Comment:
There's no need to change anything, and I'm not even sure I'm right, but I think there may be bits in the `calendar` module that may make some of this easier: https://docs.python.org/2/library/calendar.html

Only mentioning it in case it's helpful in the future :-)

Reply:
I didn't know about the calendar, thanks!  I think still some work using the calendar method.  I will leave this as-is.